test/chainrules.jl

```diff
@@ -0,0 +1,76 @@
+# Create a struct to hold the fields of the Woodbury. 
+# We do this because the Woodbury cannot represent it's own tangent. 
+# I.e. for Y = f(W,...), W̄ = ∂Y / ∂W, is not necessarily a valid Woodbury. 
+# Consider, e.g. the case of a Positive Diagonal matrix
+struct WoodburyLike
```

Comment:
I may not understand the problem enough, but could we just add a `rand_tangent` dispatch for the Woodbury which makes sure that the matrix is PSD?